docs/DeveloperGuide.md

```diff
@@ -4,26 +4,167 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Application Launch
 
+The following diagram shows a sequence diagram of the program when it is run.
+
+![ApplicationLaunchSequenceDiagram](img/ApplicationLaunchSequenceDiagram.png)
+
+1. `InvMgr` does a setup by creating the required `Ui`, `Storage`, and `ItemList` objects.
+2. To create the `ItemList` object, `load()` from `Storage` must be called. The resulting `ArrayList<Item>` from `load()` is used to create the `ItemList` object.
+3. Then, `InvMgr` will continuously loop, doing the following:
+   i. `InvMgr` will call upon `Ui` to get input from the user.
+   ii. `InvMgr` passes the user input to `Parser.parse(command)`.
+   iii. `Parser.parse(command)` returns a `Command` object.
+   iv. `InvMgr` calls upon the `execute()` method of the returned `Command` object
+4. The loop stops when the user types `exit`.
+
+### UI Component
+![UiClassDiagram](img/UiClassDiagram.png)
+
+The `UI` component consists of a single `Ui` class.
+
+The `UI` component
+- Displays salutations, prompts for user input, error messages and results of queries.
+- Reads in user inputs
+- Depends on the `Messages` and `InvMgrException` classes in the `Common` component. It displays messages stored in the `Messages` class and displays an error message whenever `InvMgrException` is invoked.
+
+### Parser Component
+
+![ParserClassDiagram](img/ParserClassDiagram.png)
+
+The diagram above shows the class diagram of how the `Parser` component works.
+
+1. The `parse(userInput)` method in `Parser` is called from `InvMgr`, and takes in the raw user input. 
+2. The method checks through a logic gate whether the user input is valid, and returns 
+a `Command` class based on the user input.
+
+### Command Component
+![CommandClassDiagram](img/CommandClassDiagram.png)
```

Comment:
You may also consider changing the way you arrange the boxes (to not in one line) to make the words larger but keep the diagram in a reasonable size.